### PR TITLE
xsdk-0.4.0: add/update package versions:

### PR DESCRIPTION
**Pull request by balay**
hypre@2.15.1
  mfem@3.5.0-rc1
  superlu-dist@6.1.0
  trilinos@12.14.0-rc1
  petsc@3.10.2
  dealii@9.0.1
  pflotran@xsdk-0.4.0
  alquimia@xsdk-0.4.0
  sundials@3.2.1
  plasma@18.11.1
  magma@2.4.0
  amrex@18.10.1
  slepc@3.10.1
  omega-h@9.19.1
  strumpack@3.1.1
  pumi@2.2.0
  tasmanian@6.0
  phist@1.7.4
  dtk-3.0 [via trilinos features '+dtk+intrepid2+shards']

add option to disable omega-h build [wrt gcc-8 conflict]

spack install xsdk~omega-h

### CODE REVIEW

**Comment by balay** [comment]
I'm submitting this PR in preparation for xsdk-0.4.0 release.

We are still in the testing process - but would like to identify and fix any potential issues with the spack changes.

cc: @jwillenbring

**Comment by tgamblin** [comment]
Marking WIP so nobody merges this -- go ahead and remove when it's ready.

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -175,7 +176,7 @@ class Trilinos(CMakePackage):
 
     resource(name='dtk',
              git='https://github.com/ornl-cees/DataTransferKit.git',
-             tag='master',
+             commit='4fe4d9d56cfd4f8a61f392b81d8efd0e389ee764',  # branch dtk-3.0
```
Comment (aprokop) [inline var/spack/repos/builtin/packages/trilinos/package.py]: I don't have a strong preference, but we could tag dtk for this release to clean this up.

Comment (aprokop) [reply var/spack/repos/builtin/packages/trilinos/package.py]: Thanks, did not know that. Then, there is little point of doing that.

**Comment by balay** [comment]
I think we are ready to push the release out - so removed the WIP label.
Also rebased this over #10166 as that is a related change.

We have one outstanding strange issue: https://github.com/dealii/dealii/issues/7543

There is another breakage with suite-sparse on ALCF/theta - which worked before - but breaking now a retest with a slightly newer compiler version [so for now I'm blaming the new compiler]

So would like to get this merged to spack now

cc: @jwillenbring @curfman

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ class Trilinos(CMakePackage):
     version('xsdk-0.2.0', tag='xsdk-0.2.0')
     version('develop', branch='develop')
     version('master', branch='master')
+    version('12.14.0-rc1', commit='dbf41f3b26b0326a1377f219e6e07eab861d181e')  # branch trilinos-release-12-14-branch
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/trilinos/package.py]: I don't normally make release candidates the default version of a package. I would set `preferred=True` on the last stable release.

**Comment by davydden** [comment]
> We have one outstanding strange issue: dealii/dealii#7543

If https://github.com/dealii/dealii/pull/7549 fixes it, we can add a patch for dealii release as a quick workaround.

**Comment by davydden** [comment]
>  the patch works!

great, i will wait until the testers are through and we merge that PR and then create a patch for Spack. Most likely the actual commit should apply without any conflicts...

**Pull request merged**